pennylane/templates/state_preparations/superposition.py

```diff
@@ -0,0 +1,302 @@
+
+        .. math::
+
+            |i\rangle \rightarrow |b_i\rangle.
+
+        Appliying these two blocks together results in the desired superposition:
+
+        .. math::
+
+            |\phi\rangle = \sum_i^m c_i |b_i\rangle.
+    """
+
+    num_wires = AnyWires
+    grad_method = None
+    ndim_params = (1,)
+
+    def __init__(
+        self, coeffs, basis, wires, work_wire, id=None
+    ):  # pylint: disable=too-many-positional-arguments, too-many-arguments
+
+        self.hyperparameters["basis"] = tuple(tuple(b) for b in basis)
+        self.hyperparameters["target_wires"] = qml.wires.Wires(wires)
+        self.hyperparameters["work_wire"] = qml.wires.Wires(work_wire)
+
+        all_wires = self.hyperparameters["target_wires"] + self.hyperparameters["work_wire"]
+
+        super().__init__(coeffs, wires=all_wires, id=id)
+
+    @property
+    def num_params(self):
+        return 1
+
+    @classmethod
+    def _primitive_bind_call(cls, *args, **kwargs):
+        return cls._primitive.bind(*args, **kwargs)
+
+    def _flatten(self):
+        metadata = tuple(
+            (key, value) for key, value in self.hyperparameters.items() if key != "target_wires"
+        )
+        return tuple(self.parameters), (self.hyperparameters["target_wires"], metadata)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        hyperparams_dict = dict(metadata[1])
+        return cls(*data, wires=metadata[0], **hyperparams_dict)
+
+    def decomposition(self):  # pylint: disable=arguments-differ
+        return self.compute_decomposition(
+            *self.parameters,
+            basis=self.hyperparameters["basis"],
+            wires=self.hyperparameters["target_wires"],
+            work_wire=self.hyperparameters["work_wire"],
+        )
+
+    @staticmethod
+    def compute_decomposition(coefs, basis, wires, work_wire):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators.
+
+        Args:
+            coefs (List[float]): List of coefficients :math:`c_i`
+            basis (List[List[int]]): List of basis states :math:`|b_i\rangle`
+            wires (Sequence[int]): List of wires that the operator acts on
+            work_wire (Union[Wires, int, str]): The auxiliary wire used for the permutation
+
+        Returns:
+            list[.Operator]: Decomposition of the operator
+
+        **Example**
+
+        .. code-block:: pycon
+
+            >>> qml.Superposition(np.sqrt([1/2, 1/2]), [[1, 1], [0, 0]], [0, 1], 2).decomposition()
+            [StatePrep(array([0.70710678, 0.70710678]), wires=[1]),
+            MultiControlledX(wires=[0, 1, 2], control_values=[False, True]),
+            CNOT(wires=[2, 0]),
+            Toffoli(wires=[0, 1, 2])]
+
+        """
+
+        dic_state = dict(zip(basis, coefs))
+        perms = _get_permutation(basis)
+        new_dic_state = {perms[key]: dic_state[key] for key in dic_state if key in perms}
+
+        sorted_coefficients = [
```

Comment:
Should we check the norm of the coefficients? or will that just happen in `StatePrep`?

Reply:
We assume that they are normalized. If not, StatePrep will display an error